src/bannerClickEvent.js

```diff
@@ -0,0 +1,29 @@
+const bannerClickEvent = () => {
+    let obj = {
+        'event': 'promotionClick',
+        'ecommerce': {
+            'promoClick': {
+                'promotions': [{
+                    'id': 'bts',
+                    'name': 'Image viewer banner',
+                    'creative': 'Banner',
+                    'position': 'On watermarked image'
+                }]
+            }
+        }
+    };
+
+    window.dataLayer = window.dataLayer || [];
+    window.dataLayer.push(obj);
+
+    return obj;
+};
+
+let banner = document.getElementById('imageviewerOverlay');
```

Comment:
Hi @punalsc - I'm not sure it's a good pattern to be mixing module definitions with applications in this way. It would be better to either:

- Separate the module definition (in this file) from it's application (which would be in a `main.js` file)
- Not use JavaScript modules at all in this instance and simply use file concatenation. 

This is introducing a couple of problems which you're having to deal with (and which you could avoid if you changed your approach):

- You're having to test for the existence of `banner` 
- You're having to test for the existence of `module` and `module.exports` because you're running this script in a browser without a transpiration step. 

I think the code could be quite a bit simpler (and more reusable) if you opt for either of the options I've mentioned above.

Reply:
Your module could look something like this if you used this approach and passed the `metaTag` element in when it's called: 

```javascript
let discoveryDetailsContentGroup = (metaTag) => {
    let metaDataContent = metaTag.getAttribute('content');

    return {
        'event': 'Discovery',
        'contentGroup': metaDataContent
    };
};

module.exports.discoveryDetailsContentGroup = discoveryDetailsContentGroup;
```

This would require an update to your tests so that it's passed in there, and you could prevent it being repeatedly defined by putting it in a scope which all tests can access:

```javascript

const metaTag = document.querySelector('meta[name=WT\\.cg_n]');


describe('Expect an object to be returned', () => {

    it('Should return an object', () => {
        expect(typeof discoveryDetailsContentGroup(metaTag)).toBe('object');
    });
});
```